src/test/java/net/openhft/chronicle/bytes/DistributedUniqueTimeProviderTest.java

```diff
@@ -0,0 +1,103 @@
+
+    @Test
+    public void currentTimeMicros() {
+        TimeProvider tp = DistributedUniqueTimeProvider.INSTANCE;
+        long last = 0;
+        for (int i = 0; i < 100_000; i++) {
+            long time = tp.currentTimeMicros();
+            assertTrue(time > last);
+            assertEquals(LongTime.toMicros(time), time);
+            last = time;
+        }
+    }
+
+    @Test
+    public void currentTimeMicrosPerf() {
+        TimeProvider tp = DistributedUniqueTimeProvider.INSTANCE;
+        long start = System.currentTimeMillis();
+        int count = 0;
+        do {
+            for (int i = 0; i < 1000; i++)
+                blackHole = tp.currentTimeMicros();
+            count += 1000;
+        } while (System.currentTimeMillis() < start + 500);
+        System.out.println("currentTimeMicrosPerf count/sec: " + count * 2);
+        assertTrue(count > 230_000 / 2); // half the speed of Rasberry Pi
+    }
+
+    @Test
+    public void currentTimeNanosPerf() {
+        TimeProvider tp = DistributedUniqueTimeProvider.INSTANCE;
+        long start = System.currentTimeMillis();
+        int count = 0;
+        do {
+            for (int i = 0; i < 1000; i++)
+                blackHole = tp.currentTimeNanos();
+            count += 1000;
+        } while (System.currentTimeMillis() < start + 500);
+        System.out.println("currentTimeNanosPerf count/sec: " + count * 2);
+        assertTrue(count > 320_000 / 2); // half the speed of Rasberry Pi
+    }
+
+    @Test
+    public void currentTimeNanos() {
+        TimeProvider tp = DistributedUniqueTimeProvider.INSTANCE;
+        long start = tp.currentTimeNanos();
+        long last = start;
+        int count = 0;
+        long runTime = Jvm.isArm() ? 3_000_000_000L : 500_000_000L;
+        for (; ; ) {
+            long now = tp.currentTimeNanos();
+            assertEquals(LongTime.toNanos(now), now);
+            if (now > start + runTime)
+                break;
+            // check the times are different after shifting by 5 bits.
+            assertTrue((now >>> 5) > (last >>> 5));
+            last = now;
+            count++;
+            if (count >= 10_000_000)
+                break;
+        }
+        System.out.printf("count: %,d%n", count);
+        assertTrue(count > 1_000_000);
+    }
+
+    @Test
+    public void concurrentTimeNanos() {
+        long start0 = System.nanoTime();
+        final int runTimeUS = 5_000_000;
+        final int threads = Jvm.isArm() ? 4 : 16;
+        final int stride = Jvm.isArm() ? 1 : threads;
+        IntStream.range(0, threads)
+                .parallel()
+                .forEach(i -> {
+                    try (DistributedUniqueTimeProvider tp = DistributedUniqueTimeProvider.forHostId(i)) {
+                        long last = 0;
+                        for (int j = 0; j < runTimeUS; j += stride) {
+                            long now = tp.currentTimeNanos();
+                            assertEquals(i, now % 100);
```

Comment:
instead of leakage of this magic number (100) - should DistributedUniqueTimeProvider have methods to extract the time and the hostId from a long nano time stamp?